src/server/api/lib/message-sending.js

```diff
@@ -392,23 +392,32 @@ export async function saveNewIncomingMessage(messageInstance) {
   let updateQuery = r.knex("campaign_contact").limit(1);
 
   if (OPT_OUT_TRIGGERS.includes(cleanedUpText)) {
-    updateQuery = updateQuery.update({ message_status: "closed" });
-
     const { id: organizationId } = await r
       .knex("organization")
       .first("organization.id")
       .join("campaign", "organization_id", "=", "organization.id")
       .join("assignment", "campaign_id", "=", "campaign.id")
       .where({ "assignment.id": assignment_id });
 
-    const optOutId = await cacheableData.optOut.save(r.knex, {
-      cell: contact_number,
-      reason: "Automatic OptOut",
-      assignmentId: assignment_id,
-      organizationId
-    });
+    const optOutQuery = r
```

Comment:
Required: We should move these checks into `optOut.save` method. We use this method in many places, and we could either add checks in every place, or just update the save method to perform the check instead.

Reply:
Looking more into this I think we have to do it in each place separately because there's other actions that also shouldn't be repeated if we already have an opt out record (ex. closing convo and queuing external sync here). I found 2 of those:

1. [`createOptout`](https://github.com/politics-rewired/Spoke/blob/main/src/server/api/root-mutations.ts#L184): My understanding is that this only gets called from a texter assignment or message review, so I think rather than check for an existing opt out here, we do `optOut.save` in a try/catch with a log. That would be similar to us logging errors when someone tries to send an initial to an opted out contact.

2. `optOut.saveMany`: To still take advantage of doing a bulk insert, I think we just modify the insert query  to add an `on conflict do nothing`. In this case we don't need to keep track of the error because we expect that users will upload numbers they've already opted out sometimes.


Does ^ sound right to you?

Reply:
Yeah. Sounds good to me! With knex 2.x you can do an `onConflict(...).update(timestamp)` so that the `updated_at` gets updated.

Reply:
Would be so nice, exceptt we're using the rethink adapter here so can't even call `onConflict()` 😞

Reply:
I suppose if `created_at` and `updated_at` have diff values we know that a user tried to manually opt someone out after they originally were. I don't feel strongly about keeping that info around. @hiemanshu were you thinking of anything else?

Reply:
> opt_out.updated_at already gets updated BEFORE UPDATE via a Postgres trigger.

If we just ignore the row, then there will be no `BEFORE UPDATE` trigger than runs to update it. 

> Question: what is the value of updating updated_at during a conflict versus just ignoring the row?

We know that the row was touched by an opt out that was triggered.